numpy/__init__.pyi

```diff
@@ -181,12 +184,114 @@ class ndarray:
     @property
     def ndim(self) -> int: ...
 
+    # Many of these special methods are irrelevant currently, since protocols
+    # aren't supported yet. That said, I'm adding them for completeness.
+    # https://docs.python.org/3/reference/datamodel.html
+    def __len__(self) -> int: ...
+    def __getitem__(self, key) -> Any: ...
+    def __setitem__(self, key, value): ...
+    def __iter__(self) -> Any: ...
+    def __contains__(self, key) -> bool: ...
+
+    def __int__(self) -> int: ...
+    def __float__(self) -> float: ...
+    def __complex__(self) -> complex: ...
+    def __oct__(self) -> str: ...
+    def __hex__(self) -> str: ...
```

Comment:
These should be conditional on the python version. I think mypy supports that?

Reply:
Yes, it supports limited version checking with `sys.version_info`.